framework/src/meshgenerators/BreakMeshByBlockGenerator.C

```diff
@@ -28,14 +28,39 @@ BreakMeshByBlockGenerator::validParams()
                              "attached. Naming convention for the new boundaries will be the old "
                              "boundary name plus \"_to_\" plus the subdomain name. At the moment"
                              "this only works on REPLICATED mesh");
+  params.addParam<std::vector<SubdomainID>>(
+      "block",
+      "The list of subdomain names where neml mehcanisc should be used, "
+      "default all blocks.");
+  params.addParam<bool>("add_transition_interface",
+                        true,
+                        "If true (default) and block is not empty, a special boundary named "
+                        "interface_transition is generate between listed blocks and other blocks. "
+                        "The transition boundary will not be split if split_interface=true");
+  params.addParam<bool>("split_transition_interface",
+                        false,
+                        "If true (default) and block is not empty, a special boundary named "
+                        "interface_transition is generate between listed blocks and other blocks. "
+                        "The transition boundary will not be split if split_interface=true");
   return params;
 }
 
 BreakMeshByBlockGenerator::BreakMeshByBlockGenerator(const InputParameters & parameters)
-  : BreakMeshByBlockGeneratorBase(parameters), _input(getMesh("input"))
+  : BreakMeshByBlockGeneratorBase(parameters),
+    _input(getMesh("input")),
+    _block(parameters.isParamSetByUser("block") ? getParam<std::vector<SubdomainID>>("block")
+                                                : std::vector<SubdomainID>(0)),
+    _block_set(_block.begin(), _block.end()),
+    _block_restricted(parameters.isParamSetByUser("block")),
+    _add_transition_interface(getParam<bool>("add_transition_interface")),
+    _split_transition_interface(getParam<bool>("split_transition_interface"))
 {
   if (typeid(_input).name() == typeid(DistributedMesh).name())
     mooseError("BreakMeshByBlockGenerator only works with ReplicatedMesh.");
+
+  if (!_add_transition_interface && _split_transition_interface)
+    mooseError("BreakMeshByBlockGenerator cannot split the transition interface becasue "
+               "add_transition_interface is false");
```

Comment:
Can you use a `paramError` here so that we get more useful output (i.e., line number):
```
paramError("split_transition_interface", "Cannot be set to true because BreakMeshByBlockGenerator cannot add (add_transition_interface = true) and split
```
or something like that?

Reply:
done